ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/Unit.hs

```diff
@@ -98,14 +98,14 @@ followerSwitchesToNewChain
 followerSwitchesToNewChain =
   let fork i = TestBody i True
   in do
-    b1 <- addBlock $ firstBlock 0 $ fork 0
-    b2 <- addBlock $ mkNextBlock b1 1 $ fork 0
+    b1 <- addBlock $ firstBlock 0 $ fork 0     -- b1 on top of G
+    b2 <- addBlock $ mkNextBlock b1 1 $ fork 0 -- b2 on top of b1
     f <- newFollower
     followerForward f [blockPoint b2] >>= \case
       Right (Just pt) -> assertEqual (blockPoint b2) pt "Expected to be at b2"
       _               -> failWith "Expecting a success"
-    b3 <- addBlock $ mkNextBlock b1 2 $ fork 1
-    void $ addBlock $ mkNextBlock b1 3 $ fork 1 -- b4
+    b3 <- addBlock $ mkNextBlock b1 2 $ fork 1  -- b3 on top of b1
+    void $ addBlock $ mkNextBlock b3 3 $ fork 1 -- b4 on top of b3
```

Comment:
Optional: We could add a check that the follower reports `AddBlock b4` after it reported `b3`. This would have caught the previously buggy implementation (as `b4` was then *not* extending `b4`).

Reply:
Bug re-introduced, before your suggestion:

```
... $ cabal run --constraint="hashable==1.4.2.0" storage-test -- -p "Follower switches to new chain"
ouroboros-storage
  Storage
    ChainDB
      Unit tests
        Follower switches to new chain
          model:  OK
          system: OK

All 2 tests passed (0.00s)
```

Bug re-introduced, after adding your suggestion:

```
... $ cabal run --constraint="hashable==1.4.2.0" storage-test -- -p "Follower switches to new chain"
ouroboros-storage
  Storage
    ChainDB
      Unit tests
        Follower switches to new chain
          model:  FAIL
            test/storage-test/Test/Ouroboros/Storage/ChainDB/Unit.hs:242:
            Expecting instruction to add a block
            Use -p '/Follower switches to new chain/&&/model/' to rerun this test only.
          system: FAIL
            test/storage-test/Test/Ouroboros/Storage/ChainDB/Unit.hs:242:
            Expecting instruction to add a block
            Use -p '/Follower switches to new chain/&&/system/' to rerun this test only.

2 out of 2 tests failed (0.00s)

... $ cabal run --constraint="hashable==1.4.3.0" storage-test -- -p "Follower switches to new chain"
ouroboros-storage
  Storage
    ChainDB
      Unit tests
        Follower switches to new chain
          model:  FAIL
            test/storage-test/Test/Ouroboros/Storage/ChainDB/Unit.hs:242:
            Rollback to wrong point
            	 Expected: At (Block {blockPointSlot = SlotNo 0, blockPointHash = TestHeaderHash (-6942406852547452771)})
            	 Actual: At (Block {blockPointSlot = SlotNo 1, blockPointHash = TestHeaderHash 2415637116744329166})
            Use -p '/Follower switches to new chain/&&/model/' to rerun this test only.
          system: FAIL
            test/storage-test/Test/Ouroboros/Storage/ChainDB/Unit.hs:242:
            Rollback to wrong point
            	 Expected: At (Block {blockPointSlot = SlotNo 0, blockPointHash = TestHeaderHash (-6942406852547452771)})
            	 Actual: At (Block {blockPointSlot = SlotNo 1, blockPointHash = TestHeaderHash 2415637116744329166})
            Use -p '/Follower switches to new chain/&&/system/' to rerun this test only.

2 out of 2 tests failed (0.00s)
```

Now fixing the bug:

```
... $ cabal run --constraint="hashable==1.4.3.0" storage-test -- -p "Follower switches to new chain"
ouroboros-storage
  Storage
    ChainDB
      Unit tests
        Follower switches to new chain
          model:  OK
          system: OK

All 2 tests passed (0.00s)
```